Formula/deno.rb

```diff
@@ -0,0 +1,76 @@
+class Deno < Formula
+  desc "command-line JavaScript / TypeScript engine"
+  homepage "https://deno.land/"
+  url "https://chromium.googlesource.com/chromium/tools/depot_tools.git",
+      :revision => "aec259ea62328ce39916607876956239fbce29b8"
+  version "0.3.3" # the version of the deno checkout, not a depot_tools version
+
+  # depot_tools/GN require Python 2.7+
+  # depends_on "python@2" => :build
+
+  depends_on "node" => :build
+  depends_on "rust" => :build
+
+  # https://bugs.chromium.org/p/chromium/issues/detail?id=620127
+  depends_on :macos => :el_capitan
+
+  def install
+    # Add depot_tools in PATH
+    ENV.prepend_path "PATH", buildpath
```

Comment:
```suggestion
    resource("gn").stage do
      system "python", "build/gen.py"
      system "ninja", "-C", "out/", "gn"
    end
    ENV.prepend_path "PATH", buildpath/"gn/out"
```

and add a resource block pointing to the lastest `gn` commit

Reply:
@chrmoritz you should be able to push to this branch now. Please can you try with deno 0.5.0 and we'll see if we can debug?

Reply:
@hayd Thanks, I've pushed my latest version based on `deno` 0.5.0 to this PR and rebased it on `master` (so that we can test with latest `rust`...).

Reply:
ry suggested on gitter it could be that node_modules isn't symlinked to third_party/node_modules. This is one of the things that setup.py/build.py does this I think (but can't see where).

In early versions of this formula I remember having to do strange things with that.
https://github.com/Homebrew/homebrew-core/pull/35645/commits/267aadb894ff384e206c6236b307d7b430d0d596

Reply:
You probably mean that symlinking which is done in [`run_node.py`](https://github.com/denoland/deno/blob/v0.5.0/tools/run_node.py). I think `gn` is calling this script just fine and the `rollup` script which is called with it seems to run just fine. It's the snapshot creation which fails for the `compiler.js`, but just works fine for the `main.js`.

Reply:
There is definitely something wrong with the generated `compiler.js`. Here is the diff between a `compiler.js` from a working build outside homebrew to a broken one within homebrew: https://gist.github.com/chrmoritz/49ef50d3e1b54ca4a64375eafa6ff1e3

Reply:
I've fixed it with patching the rollup config with
```
inreplace "rollup.config.js", "third_party/node_modules/typescript/lib/typescript.js", "node_modules/typescript/lib/typescript.js"
```
But to be honest I don't know why rollup is using the `node_modules` symlink in the root directory for a homebrew build, but the actual `node_modules` inside `third_party` outside homebrew. I've just looked at the symptoms of the failed build and adjusted the `rollup.config.js` according to it.

Reply:
Wow, well done! Yeah, this definitely feels weird.

Reply:
Maybe the symlink isn't working correctly?

I remember this was an issue in this commit https://github.com/Homebrew/homebrew-core/pull/35645/commits/548d280656b246b5169bff27692e16c999f8e778 (perhaps it's still unresolved, or this rm was moved to a python file that's no longer called?)

Reply:
The symlink is actual required for the TypeScript transpilation to work (I've already tried to remove it). If we remove the symlink we need to [patch this line](https://github.com/denoland/deno/blob/ec9080f34c936d9af56cca68de664954053bf423/cli/BUILD.gn#L203) to `rebase_path("//third_party/node_modules/ts-node/dist/bin.js", root_build_dir),`

Reply:
Previously symlinks needed, see commit, to be removed before anything else (a bug I couldn't recreate locally, only on jenkins... seemingly the git clone of the [symlinks](https://github.com/denoland/deno/blob/master/node_modules) screws up) i.e. try the rm before running gn etc. ?

Reply:
Oh, your right. `brew` seems to replace all symlinks with actual copies of the folders and I can reproduce this locally. I wonder if this is intended or a bug in `brew`.